packages/server/lib/server.js

```diff
@@ -131,6 +131,13 @@ class Server {
     const { morgan, clientRoute } = config
     const app = express()
 
+    app.use(function (req, res, next) {
+      req.pipe(concatStream((reqBody) => {
+        req.rawBody = reqBody.toString()
+        next()
+      }))
+    })
```

Comment:
as you've probably noticed, this breaks the request because the request body stream is drained

there is already some code in intercept-request that buffers the request body: https://github.com/cypress-io/cypress/blob/1aa5112086234ffe279e9f972f385c2e75cfa2e5/packages/net-stubbing/lib/server/intercept-request.ts#L203-L213

req.body is specially handled too, if it is present, it is used instead of the `req` `ReadableStream` when sending the outgoing request:

https://github.com/cypress-io/cypress/blob/1aa5112086234ffe279e9f972f385c2e75cfa2e5/packages/proxy/lib/http/request-middleware.ts#L111-L111

https://github.com/cypress-io/cypress/blob/1aa5112086234ffe279e9f972f385c2e75cfa2e5/packages/proxy/lib/http/request-middleware.ts#L121-L123

https://github.com/cypress-io/cypress/blob/1aa5112086234ffe279e9f972f385c2e75cfa2e5/packages/proxy/lib/http/request-middleware.ts#L134-L137

so you don't need to introduce a new `rawBody` property, `req.body` is already good enough

Reply:
I had to create `rawBody` property because `body` property is unavailable at `_getRouteForRequest` because `request` is buffered at `_interceptRequest` which is called later than `_getRouteForRequest`. 

To solve this problem, I also drained the buffer earlier to `req.body` inside the middleware above. But it also failed because buffers are not drained when `staticResponse` is not undefined. Because of that, draining earlier for all `route2` calls fails to send response to the browser in calls like `cy.route2({ url: 'https://test.com', body: '1234' }, 'numbers' })`. 

It seems that we might need another solution.